com.unity.netcode.gameobjects/Runtime/Spawning/NetworkSpawnManager.cs

```diff
@@ -422,6 +422,12 @@ internal void SpawnNetworkObjectLocally(NetworkObject networkObject, ulong netwo
                 throw new SpawnStateException("Object is already spawned");
             }
 
+            var networkObjectChildren = networkObject.GetComponentsInChildren<NetworkObject>();
+            if (networkObjectChildren.Length > 1)
+            {
+                Debug.LogError("Spawning NetworkObjects with nested NetworkObjects is not supported and child NetworkObjects will not be spawned over the network!");
+            }
+
```

Comment:
this block should live under an "if not scene object" check, otherwise in-scene networkobject reparenting would also throw this error. in fact, that's what's failing testproject tests atm.

Reply:
Updated, see latest commit.